### PR TITLE
Feature/tag

### PR DESCRIPTION
**Pull request by LCarrasquel14**
added styles I need to place the line below and when clicking it calls handleBoardView and makes a console log showing the id that we passed through prot

### CODE REVIEW

File: src/Components/bar.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import clsx from "clsx";
+
+type Props = {
+  progress: number;
+  maxLimit: number;
+  validate?: boolean;
+};
+
+const percentageBar = ({ progress, maxLimit, validate }: Props) => {
```
Comment (enmaGit) [inline src/Components/bar.tsx]: Name of the components should start with capital case

File: src/app/layout.tsx [hunk 2]
```diff
@@ -12,7 +12,15 @@ export default function RootLayout({
 }>) {
   return (
     <html lang="en">
-      <body>{children}</body>
+      <body>
+        {children}
+        <script
+          src="https://cdnjs.cloudflare.com/ajax/libs/react-modal/3.14.3/react-modal.min.js"
```
Comment (enmaGit) [inline src/app/layout.tsx]: We need to remove this react-modal script. This needs to be installed by npm not by the cdn

File: src/ui/Modal/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  isOpen: boolean;
+  onClose?: () => void;
+  children?: ReactNode;
+};
+
+const Modal: FC<Props> = ({ isOpen, onClose, children }) => {
+  return (
+    <div>
+      {isOpen && (
+        <i
```
Comment (enmaGit) [inline src/ui/Modal/Modal.tsx:16]: Why is this open separated from the modal?

File: src/ui/Tag/Tag.tsx [hunk 4]
```diff
@@ -15,21 +13,25 @@ type TagProps = {
   tagContainer: tagProps[];
 };
 
-const Tag = ({ tagContainer = tags }: TagProps) => {
+const Tag = ({ tagContainer }: TagProps) => {
```
Comment (enmaGit) [inline src/ui/Tag/Tag.tsx]: It is not Tag but Tab

File: src/ui/Toggle/Toggle.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  buttons: [Button, Button];
+};
+
+const Toggle = ({ buttons }: Props) => {
+  const [selected, setSelected] = useState(buttons[0].selected);
+  const onToggle = () => {
+    buttons.map((button) => {
+      if (button.selected) {
+        setSelected(false);
+        button.selected = false;
```
Comment (enmaGit) [inline src/ui/Toggle/Toggle.tsx:22]: You are mutating the props and this is not safe. You shouldn't mutate the props, you should change the state and then comunicate to the parent through a callback that the input changed.

**Review by enmaGit — APPROVED** [review]

File: src/ui/Toggle/Toggle.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  label: string;
+  icon: IconProps["icon"];
+  id: string;
+  selected: boolean;
+};
+type Props = {
+  buttons: [Button, Button];
+};
+
+const Toggle = ({ buttons }: Props) => {
```
Comment (enmaGit) [inline src/ui/Toggle/Toggle.tsx:16]:
```suggestion
const Toggle = ({ buttons, onToggle }: Props) => {
```

File: src/ui/Toggle/Toggle.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  const [selected, setSelected] = useState(buttons[0].selected);
+  const onToggle = () => {
+    buttons.map((button) => {
```
Comment (enmaGit) [inline src/ui/Toggle/Toggle.tsx:19]:
```suggestion
    buttons.forEach((button) => {
        setSelected(selected === 1 ? 0 : 1);
        onToggle();
    }
```

File: src/ui/Modal/Modal.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      <ReactModal
+        isOpen={isOpen}
+        overlayClassName="modal-overlay"
+        className="modal-content"
+        aria={{
+          labelledby: "heading",
+          describedby: "full_description",
+        }}
+      >
+        {children}
```
Comment (enmaGit) [inline src/ui/Modal/Modal.tsx:32]:
```suggestion
    <>
         <i></i>
           {children}
     </>
```

**Review by enmaGit — CHANGES_REQUESTED** [review]

File: src/Components/bar.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import clsx from "clsx";
+
+type Props = {
+  progress: number;
+  maxLimit: number;
+  validate?: boolean;
+};
+
+const PercentageBar = ({ progress, maxLimit, validate }: Props) => {
```
Comment (enmaGit) [inline src/Components/bar.tsx]: We need to remove the validte from here, any validation logic for this should be inside the component

File: src/ui/Progress/Progress.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+import ProgressBar from "@/Components/Bar";
+
+type Props = {
+  maxLimit: number;
+  progress: number;
+};
+
+const Progress = ({ maxLimit, progress }: Props) => {
+  const [showMenu, setShowMenu] = React.useState(false);
+  const validate = progress <= maxLimit;
```
Comment (enmaGit) [inline src/ui/Progress/Progress.tsx]: This logic needs to be inside the ProgressBar

**Review by enmaGit — CHANGES_REQUESTED** [review]

**Comment by LCarrasquel14** [comment]
squash and merge

**Pull request merged**